### DISCUSSION TITLE
How to simulate multiple client applications?

### DISCUSSION THREAD

**Discussion by ZedZipDev** [xunit/xunit#2443] [Question] [answered]
I have a NET Core service with Rest API. In the same solution I have added xunit test Project with some test set.
It works fine. 
The next question appeared: I need to test this service with simultaneous multiple client connections. Is it possible to simulate it in xunit? If yes - how to?

**Comment by bartelink** [comment]
Can you expand on what you are talking about. IIRC I saw you ask a similar question on SO - bottom line is you ned to type more than a line or two to explain what you're after if people are going to be able to give you a reasonable answer aside from splurging out some massive high level testing strategies. I'd suggest closing this and asking a more detailed question on StackOverflow.com - anyone that watches here also watches the `xunit` tag on SO.

**Comment by bradwilson** [comment, marked as answer]
While you could do this manually (create a bunch of worker threads that spawn clients, collect their results, etc.) it's not something that's built-in (and frankly not something that this framework was designed for). Proper performance/safety measurement and validation is a separate problem space that could probably use a separate tool to achieve.
